app/components/ocl_tools/date_picker_component.rb

```diff
@@ -18,8 +19,13 @@ def initialize(ids:, names:, placeholder:, value: nil, start_year: nil, end_year
 
       classes = "shadow-sm focus:ring-indigo-500 focus:border-indigo-500 inline w-full sm:text-sm border-gray-300 rounded-md".freeze
       error_classes = "shadow-sm block w-full sm:text-sm rounded-md border-red-300 text-red-900 placeholder-red-900 focus:outline-none focus:ring-red-500 focus:border-red-500".freeze
+      disabled_classes = "shadow-sm cursor-not-allowed focus:outline-none focus:ring-gray-200 focus:border-gray-300 inline w-full sm:text-sm border-gray-300 rounded-md".freeze
 
-      @classes = errors ? error_classes : classes
+      @classes = if disabled
+                   disabled_classes
+                 else
+                   (errors ? error_classes : classes)
+                 end
```

Comment:
I'm keen to see classes in the component if possible, so that all the information affecting the display is in one place. In this case it would be pretty easy to do this, as `@classes` is only used on one line. Something like:
```html
<% if @disabled %>
  ...
<% elsif @errors %>
 ...
<% else %>
...
<% end %>
```
Would you be open to making this change?

Reply:
Done